src/sonic-ctrmgrd/ctrmgr/container_startup.py

```diff
@@ -0,0 +1,293 @@
+    key = _get_version_key(feature, version)
+    return (key in labels) and (labels[key].lower() == "false")
+
+
+def drop_label(feature, version):
+    # Mark given feature version as dropped in labels.
+    # Update is done in state-db.
+    # ctrmgrd sets it with kube API server per reaschability
+    
+    tbl = swsscommon.Table(state_db, KUBE_LABEL_TABLE)
+    name = _get_version_key(feature, version)
+    tbl.set(KUBE_LABEL_SET_KEY, [ (name, "false")])
+        
+
+def update_data(feature, data):
+    # Update STATE-DB entry for this feature with given data
+    #
+    debug_msg("{}: {}".format(feature, str(data)))
+    tbl = swsscommon.Table(state_db, "FEATURE")
+    tbl.set(feature, list(data.items()))
+
+
+def get_docker_id():
+    # Read the container-id
+    # Note: This script runs inside the context of container
+    #
+    cmd = 'cat /proc/self/cgroup | grep -e ":memory:" | rev | cut -f1 -d\'/\' | rev'
+    proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+    output = proc.communicate()[0].decode("utf-8")
+    return output.strip()[:12]
+
+
+def instance_higher(feature, version):
+    # Compares given version against current version in STATE-DB.
+    # Return True if this version is higher than current.
+    #
+    ret = False
+    ct_version = state_data[VERSION]
+    if ct_version:
+        ct = ct_version.split('.')
+        nxt = version.split('.')
+        for cs, ns in zip(ct, nxt):
+            c = int(cs)
+            n = int(ns)
+            if n < c:
+                break
+            elif n > c:
+                ret = True
+                break
+    else:
+        # Empty version implies no one is running.
+        ret = True
+
+    debug_msg("compare version: new:{} current:{} res={}".format(
+        version, ct_version, ret))
+    return ret
+
+
+def is_active(feature):
+    # Check current system state of the feature
+    if state_data[SYSTEM_STATE] == "up":
+        return True
+    else:
+        syslog.syslog(syslog.LOG_ERR, "Found inactive for {}".format(feature))
+        return False
+
+
+def update_state(feature, owner=None, version=None):
+    """
+    sets owner, version & container-id for this feature in state-db.
+
+    If owner is local, update label to block remote deploying same version or
+    if kube, sets state to "running". 
+
+    """
+    data = {
+            CURRENT_OWNER: owner,
+            DOCKER_ID: get_docker_id() if owner != "local" else feature,
+            UPD_TIMESTAMP: str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")),
+            "hello": "world",
+            VERSION: version
+            }
+
+    if (owner == "local"):
+        # Disable deployment of this version as available locally
+        drop_label(feature, version)
+    else:
+        data[REMOTE_STATE] = "running"
+
+    debug_msg("{} up data:{}".format(feature, str(data)))
+    update_data(feature,  data)
+    state_data.update(data)
+
+
+def do_exit(feat, m):
```

Comment:
the function name is a little bit confusing, why is this call do_exit() but somehow from the fucntion it does the sleep and will not return.

Reply:
s/do_exit/do_freeze/